### ISSUE TITLE
Calling a method through a ParamSpec self-type rejects every argument

### ISSUE THREAD

**Issue by jvdillon** [astral-sh/ty#4352]
### Summary

ty 0.0.73 rejects a call that a gradual signature should accept. 0.0.72 and basedpyright 1.39.10 work correctly.

Example:

```python
from typing import Any, Protocol

class HasForward[**P, R](Protocol):
    def forward(self, *args: P.args, **kwargs: P.kwargs) -> R: ...

class Mod:
    def __call__[**P, R](self: HasForward[P, R], *a: P.args, **k: P.kwargs) -> R:
        return self.forward(*a, **k)

    def forward(self, *args: Any, **kwargs: Any) -> Any:
        return (args, kwargs) 

Mod()(1)
```

produces,

```
warning[invalid-argument-type]: Argument to bound method `Mod.__call__` is incorrect
  --> a.py:13:7
   |
13 | Mod()(1)
   |       ^ Expected `(...)`, found `Literal[1]`
info: Method defined here
 --> a.py:7:9
  |
7 |     def __call__[**P, R](self: HasForward[P, R], *a: P.args, **k: P.kwargs) -> R:
  |         ^^^^^^^^                                 ---------- Parameter declared here

Found 1 diagnostic
```

### Version

0.0.73 and 0.0.74

**Comment by carljm** [comment]
Thanks!
